RoddyCore/test/de/dkfz/roddy/config/ToolFileParameterTest.groovy

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2017 eilslabs.
+ *
+ * Distributed under the MIT License (license terms are at https://www.github.com/eilslabs/Roddy/LICENSE.txt).
+ */
+
+package de.dkfz.roddy.config
+
+import de.dkfz.roddy.knowledge.files.BaseFile
+import groovy.transform.CompileStatic;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+/**
+ * Created by heinold on 08.03.17.
+ */
+@CompileStatic
+public class ToolFileParameterTest {
+    @Test
+    public void testCloneAndEquals() throws Exception {
+        def src = new ToolFileParameter(BaseFile, [], "ABC", new ToolFileParameterCheckCondition(true), "DEF", null, "abc")
+        def clone = src.clone()
+        assert clone == src
```

Comment:
??? Why are you testing `==` and not `.equals()`? Thus src.clone() could/should return `this`. Also, the issue with the `scriptparameter` can not be found. There is also no negative controls here! You'd need to test with .equals() and have tests in which each one of the values is different.